tools/README.rst

```diff
@@ -22,170 +13,57 @@ Overall, the conformance test tools work like this:
-
-This converts a source ``.textproto`` file to an interim file, and from there to a final ``.feature`` file.
-
-A good way to use this is to do a checkout from https://github.com/google/cel-spec.git into
-an adjacent directory. By deafult, the Makefile looks for ``../../cel-spec``. If this doesn't work
-for you, set the ``CEL_SPEC_PATH`` environment variable.
-
-On Gherkinization
-=================
-
-The question arises on how best to serialize descriptions of the objects created by a CEL evaluation.
-We have three distinct notations to deal with:
-
--   Source Protobuf text in the ``.textproto`` files.
-
--   Final Python object used by Behave.
-
--   An intermediate text representation in the Gherkin that's created by a Go application.
-
-The idea is to write Gherkin tests that match the protobuf source closely, but can be processed by Behave without too much overhead.
-(There are over 900 conformance tests; we don't want to take all day.)
-
-The use of a **Go** application is far simpler than trying to parse the ``.textproto`` files in Python.
-The **Go** operations for expanding each ``.Section`` and ``.Test`` object into a Gherkin Scenario is very small, and narrowly focused on creating an easier-to-parse representation.
-
-There are three broad choices for Gherkin representation of expected results.
-
--   Protobuf, unmodified.  ``{ int64_value: -1 }``.
-    This pushes the final parsing into the conformance test step definitions.
+    % python ../tools/gherkinize.py basic.textproto -o basic.feature
 
--   Text representation of the Python target object. ``celpy.celtypes.IntType(-1)``.
-    This is highly Python-specific and not of general use to other implementers.
+This parses a source ``.textproto`` file and generates an equivalent ``.feature`` file.
 
--   An intermediate representation. ``Value(value_type='int64_value', source='x')``.
-    This is trivially produced by **Go**.
-    It preserves the protobuf semantics in a easy-to-parse form.
+A good way to use this is to do a checkout from https://github.com/google/cel-spec into an adjacent
+directory. By default, the Makefile looks for ``<repo>/../../google/cel-spec`` but the location can
+be overridden with the ``CEL_SPEC_PATH`` environment variable.
 
-The Python representation leverages the following class definition
+gherkinize.py
+=============
 
-..  code-block:: python
+Each ``.textproto`` file follows the ``SimpleTestFile`` `schema`_.
 
-    class Primitive(NamedTuple):
-        """A name: value pair."""
-        type_name: Token
-        value_text: Token
+.. _schema: https://buf.build/google/cel-spec/docs/main:cel.expr.conformance.test#cel.expr.conformance.test.SimpleTestFile
 
-        @property
-        def value(self) -> Any:
-            ...
+These are deserialized into Python objects using an `SDK generated from the schemas`_.
 
+.. _SDK generated from the schemas: https://buf.build/google/cel-spec/sdks/dffb8c8cf7814e96a7c06b6a5ac72fdc:protocolbuffers/python
```

Comment:
may want to also use `main` here so the link doesn't go out of date as the versions are upgraded?
```suggestion
.. _SDK generated from the schemas: https://buf.build/google/cel-spec/sdks/main:protocolbuffers/python
```